web/src/recent_view_util.ts

```diff
@@ -14,14 +12,25 @@ export function get_topic_key(stream_id: number, topic: string): string {
     return stream_id + ":" + topic.toLowerCase();
 }
 
-export function get_key_from_message(msg: Message): string {
-    if (msg.type === "private") {
+export function get_key_from_conversation_data(
+    data:
+        | {
+              type: "private";
+              to_user_ids: string;
+          }
+        | {
+              type: "stream";
+              stream_id: number;
+              topic: string;
+          },
+): string {
```

Comment:
Should this instead just take a `ConversationData`? Not sure why you're defining this other abbreviated type here.

Reply:
It's because we call it in `process_message` with a message:

`const key = get_key_from_conversation_data(msg);`

and `ConversationData` has attributes that aren't on `Message` which also aren't needed here.